### PR TITLE
Grafana-UI: Enable empty time range

### PR DESCRIPTION
**Pull request by Clarity-89**
**What this PR does / why we need it**:
Add a possibility to use `TimeRangeInput` in an empty state, without initial date value as well as enabling clearing the input.

Screenshot 2020-07-14 at 12 40 10

Screenshot 2020-07-14 at 12 40 18

### CODE REVIEW

File: packages/grafana-ui/src/components/TimePicker/TimeRangeInput.tsx [hunk 1]
```diff
@@ -45,19 +66,34 @@ export const TimeRangeInput: FC<Props> = ({
     onChange(timeRange);
   };
 
+  const onRangeClear = (event: MouseEvent<HTMLDivElement>) => {
+    event.stopPropagation();
+    const from = '';
+    const to = '';
+    onChange({ from, to, raw: { from, to } });
```
Comment (dprokop) [inline packages/grafana-ui/src/components/TimePicker/TimeRangeInput.tsx:73]: I think we should allow `dateTime(null)`. moment  accepts the null values producing date object that is invalid (isValid === false). Ref https://stackoverflow.com/a/57817236

**Review by dprokop — COMMENTED** [review]
Left a few comments. I'm mostly concerned with `InputTimeRange` interface that feels unnecessary and would love to get rid of it.

**Comment by dprokop** [comment]
UX/UI wise this looks great!

File: packages/grafana-ui/src/components/TimePicker/TimeRangeInput.tsx [hunk 2]
```diff
@@ -9,13 +9,32 @@ import { getFocusStyle } from '../Forms/commonStyles';
...
+  from: dateTime().subtract(6, 'hour'),
+  to: dateTime(),
+  raw: { from: 'now-6h', to: 'now' },
+};
+
+export interface InputTimeRange {
+  from: DateTime | string;
+  to: DateTime | string;
+  raw: RawTimeRange;
+}
```
[the remark this answers is not in the document]
Comment (Clarity-89) [reply packages/grafana-ui/src/components/TimePicker/TimeRangeInput.tsx:24]: We need to have some kind of empty input value for the time range. For example in the report form if range is not specified that means that custom time range isn't used. The current `TimeRange` interface doesn't allow empty time ranges, so I had to make a separate on for the input, where an empty string in the range would signify no value. 

Agree, it doesn't seem like the cleanest solution but I couldn't figure out a better way to do it.

File: packages/grafana-ui/src/components/TimePicker/TimeRangeInput.tsx (hunk 1 above)
Comment (Clarity-89) [reply packages/grafana-ui/src/components/TimePicker/TimeRangeInput.tsx:73]: Yeah, that seems like a good option. It's just the api often stores time values in a string and returns empty strings when there's no value.

Comment (Clarity-89) [reply packages/grafana-ui/src/components/TimePicker/TimeRangeInput.tsx:73]: Ok, now I see what you mean, indeed seems like `dateTime(null)` is working, so I can get rid of `InputTimeRange` interface. Just need to mention in the docs that `dateTime(null)` should be provided for time range values.

File: packages/grafana-data/src/datetime/moment_wrapper.ts [hunk 3]
```diff
@@ -5,7 +5,7 @@ export interface DateTimeBuiltinFormat {
   __momentBuiltinFormatBrand: any;
 }
 export const ISO_8601: DateTimeBuiltinFormat = moment.ISO_8601;
-export type DateTimeInput = Date | string | number | Array<string | number> | DateTime; // null | undefined
+export type DateTimeInput = Date | string | number | Array<string | number> | DateTime | null; // | undefined;
```
Comment (Clarity-89) [inline packages/grafana-data/src/datetime/moment_wrapper.ts:8]: Had to uncomment this to allow null values, no idea why it was commented out in the first place.

Comment (dprokop) [reply packages/grafana-data/src/datetime/moment_wrapper.ts:8]: Yeah, no idea neither, but it should be OK

File: packages/grafana-ui/src/components/TimePicker/TimeRangeInput.tsx (hunk 1 above)
Comment (dprokop) [reply packages/grafana-ui/src/components/TimePicker/TimeRangeInput.tsx]: perfect!

**Review by dprokop — APPROVED** [review]
@Clarity-89 looks great, awesome that you got rid of that specific interface for clearable values. Awesome job!

**Pull request merged**